File: bindings/python/tests/identify.py
import unittest

import sys
# automake build dir
sys.path.insert(0, '..')
sys.path.insert(0, '../.libs')
# cmake build dir
sys.path.insert(0, '../../../build/bindings/python')

from pywsman import *



class TestSequenceFunctions(unittest.TestCase):

   def test_identify(self):
#     set_debug(1) # enable to print logging to stderr
      client = Client( "http://wsman:secret@localhost:5985/wsman" )
      assert client is not None
      client.transport().set_auth_method(BASIC_AUTH_STR) # Windows winrm needs this
      options = ClientOptions()
      assert options is not None
#     options.set_dump_request() # enable to print SOAP request to stdout
      doc = client.identify( options )
      assert doc is not None
#      print "Document [%s]" % doc
      root = doc.root()
      assert root is not None
#      print "Root [%s]" % root
      prot_version = root.find( XML_NS_WSMAN_ID, "ProtocolVersion" )
      prod_vendor = root.find( XML_NS_WSMAN_ID, "ProductVendor" )
      prod_version = root.find( XML_NS_WSMAN_ID, "ProductVersion" )
      print "Protocol %s, Vendor %s, Version %s"  %( prot_version, prod_vendor, prod_version )


if __name__ == '__main__':
    unittest.main()

File: bindings/perl/tests/identify.pl
#!/usr/bin/perl

#
# perl example to get openwsmand info.
# written by warptrosse@gmail.com
#

use strict;
use warnings;
use lib '../../../build/bindings/perl';
use lib '../.libs';
use lib '..';
use openwsman;

# debug (to stderr)
# openwsman::set_debug(-1);

# Create client instance.
# (host, port, path, scheme, username, password)
my $client = new openwsman::Client::('localhost', 5985, '/wsman', 'http',
                                     'wsman', 'secret')
    or die print "[ERROR] Could not create client handler.\n";

# Alternate way.
# my $client = new openwsman::Client::('http://wsman:secret@localhost:8889/wsman')
#  or die print "[ERROR] Could not create client handler.\n";

# Set up client options.
my $options = new openwsman::ClientOptions::()
    or die print "[ERROR] Could not create client options handler.\n";

# Dump the XML request to stdout.
# $options->set_dump_request();

# Force basic auth.
$client->transport()->set_auth_method($openwsman::BASIC_AUTH_STR);

my $result; # Used to store obtained data.

# Identify.
# (options)
$result = $client->identify($options)
unless($result && $result->is_fault eq 0) {
    die print "[ERROR] Could not identify server.\n";
}

# Get server info.
my $root = $doc->root;
my $prot_version = $root->find($openwsman::XML_NS_WSMAN_ID,
                               "ProtocolVersion")->text();
my $prod_vendor = $root->find($openwsman::XML_NS_WSMAN_ID,
                              "ProductVendor")->text();
my $prod_version = $root->find($openwsman::XML_NS_WSMAN_ID,
                               "ProductVersion")->text();

# Print output.
print "Protocol $prot_version, Vendor $prod_vendor, Version $prod_version\n";
